websites/T/TuneIn/presence.ts

```diff
@@ -1,85 +1,117 @@
+				if (title.length > 128)
+					presenceData.details = `${title.substring(0, 125)}...`;
+				else presenceData.details = title;
+			}
+
+			if (author) {
+				if (author.length > 128)
+					presenceData.state = `${author.substring(0, 125)}...`;
+				else presenceData.state = author;
+			}
 
-			presenceData.state = author;
-			if (author.length > 128)
-				presenceData.state = `${author.substring(0, 125)}...`;
-			if (pauseCheck[0]) {
+			if (artwork && cover) presenceData.largeImageKey = artwork;
+		}
+
+		if (conditions.isLive.textContent === "") {
+			if (conditions.isStopped) {
+				delete presenceData.startTimestamp;
 				presenceData.smallImageKey = Assets.Pause;
 				presenceData.smallImageText = (await strings).pause;
 			} else {
-				presenceData.smallImageKey = "live";
+				if (!private && timestamps) {
+					const timeElapsed = presence.timestampFromFormat(
+						document
+							.querySelector("#scrubberElapsed")
+							.getAttribute("data-elapsedtime")
+					);
+					presenceData.startTimestamp =
+						Math.floor(Date.now() / 1000) - timeElapsed;
+				}
+				presenceData.smallImageKey = Assets.Live;
 				presenceData.smallImageText = (await strings).live;
 			}
 		} else {
-			title = document.querySelector("#playerTitle").textContent;
-			author = document.querySelector("#playerSubtitle").textContent;
-			const timestamps = presence.getTimestamps(
-					presence.timestampFromFormat(
-						document.querySelector("#scrubberElapsed").textContent
-					),
-					presence.timestampFromFormat(
-						document.querySelector("#scrubberDuration").textContent
-					)
-				),
-				paused = document
-					.querySelector("#innerAppContent")
-					.querySelectorAll('[data-testid="player-status-paused"]');
+			if (!private && timestamps) {
+				const elapsed = document.querySelector("#scrubberElapsed").textContent,
+					duration = document.querySelector("#scrubberDuration").textContent;
 
-			presenceData.details = title;
-			if (title.length > 128)
-				presenceData.details = `${title.substring(0, 125)}...`;
+				if (elapsed !== "00:00" || duration !== "") {
+					const timestamps = presence.getTimestamps(
+						presence.timestampFromFormat(elapsed),
+						presence.timestampFromFormat(duration)
+					);
 
-			presenceData.state = author;
-			if (author.length > 128)
-				presenceData.state = `${author.substring(0, 125)}...`;
+					presenceData.startTimestamp = timestamps[0];
+					presenceData.endTimestamp = timestamps[1];
+				}
+			}
 
-			if (paused[0]) {
+			if (conditions.isPaused) {
 				delete presenceData.startTimestamp;
 				delete presenceData.endTimestamp;
 				presenceData.smallImageKey = Assets.Pause;
 				presenceData.smallImageText = (await strings).pause;
+			} else if (conditions.isStopped) {
+				presenceData.smallImageKey = Assets.Stop;
+				presenceData.smallImageText = (await strings).pause;
 			} else {
 				presenceData.smallImageKey = Assets.Play;
 				presenceData.smallImageText = (await strings).play;
```

Comment:
With the current `strings` implementation, there is no longer a need to `await` strings, as it is awaited at the top and should no longer be a `Promise`

Reply:
> With the current `strings` implementation, there is no longer a need to `await` strings, as it is awaited at the top and should no longer be a `Promise`

Got it, removed the `await` from strings! 👍